flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/optimize/ClearQueryBlockAliasResolver.java

```diff
@@ -31,22 +31,19 @@
 import java.util.stream.Collectors;
 
 /** A shuttle to remove query block alias hint. */
-public class ClearQueryBlockAliasResolver extends RelShuttleImpl {
+public class ClearQueryBlockAliasResolver extends RelHomogeneousShuttle {
 
     public List<RelNode> resolve(List<RelNode> roots) {
         return roots.stream().map(node -> node.accept(this)).collect(Collectors.toList());
     }
 
     @Override
-    protected RelNode visitChild(RelNode parent, int i, RelNode child) {
-        RelNode newParent = clearQueryBlockAlias(parent);
-        return super.visitChild(newParent, i, child);
-    }
-
-    @Override
-    public RelNode visit(TableScan scan) {
-        RelNode newScan = clearQueryBlockAlias(scan);
-        return super.visit(newScan);
+    public RelNode visit(RelNode other) {
+        if (FlinkRelOptUtil.containsSubQuery(other)) {
+            other = FlinkHints.resolveSubQuery(other, relNode -> relNode.accept(this));
+        }
+        RelNode result = clearQueryBlockAlias(other);
```

Comment:
Why the issue cannot be fixed by adjusting the `clearQueryBlockAlias` method?

Reply:
Because I want let `clearQueryBlockAlias` have a single responsibility, that is, given a RelNode and try to clear unreasonable hints on this relnode. `ClearQueryBlockAliasResolver` itself is responsible for traversing all nodes.
WDYT?

Reply:
So, the original problem was that while traversing, we were not able to reach the `RelNode` that had query block alias?

Reply:
Yes. It's not that it is not able to be done; it's just that it hasn't been done. So I added this piece of logic. 
```
if (FlinkRelOptUtil.containsSubQuery(other)) {
      other = FlinkHints.resolveSubQuery(other, relNode -> relNode.accept(this));
}
```